File: Hackerrank/bfsshortreach.cpp
#include <iostream>
#include <vector>
#include <map>
#include <set>
#include <queue>

using namespace std;


void bfs(map< int, set<int> > graph, int primaryNode, int level[1005]){
    int visited[1005]={0};
    queue<int> q;
    q.push(primaryNode);
    level[primaryNode]=0;
    while(!q.empty()){
         int primaryNode = q.front();
         q.pop();
           for(std::set<int>::iterator it = graph[primaryNode].begin(); it!=graph[primaryNode].end(); it++){
                  if(visited[*it]==0){
                             q.push(*it);
                             level[*it]=depth;
    			     visited[*it]=1;                        
     			 }
                  }
            visited[primaryNode]=1;
         }
    
}

int main(){
 int t,n,m; cin>>t;

 while(t--){
   int level[1005];
   map <int, set<int> > graph;
   int node1, node2, primaryNode;

   cin>>n>>m;
   for(int i=0; i<m; i++){
     cin>>node1>>node2;
     graph[node1].insert(node2);
     graph[node2].insert(node1);
   }

   cin>>primaryNode;
   bfs(graph,primaryNode,level);

   for(int i=1;i<=n;i++)
      if(i!=primaryNode){
        if(level[i]==0)cout<<-1<<" ";
        else
          cout<<level[i]*6<<" ";
      }
      cout<<"\n";
 }

}

File: Hackerrank/bfsshortreach.py
t=input()

for _ in xrange(t):
    n,m = map(int,raw_input().split())
    graph = [set() for i in xrange(n+1)]
    for i in xrange(m):
        a,b=map(int,raw_input().split())
        graph[a].add(b)
        graph[b].add(a)
    start = input()

    q=[start]
    level=[-1]*(n+1)
    visited=[0]*(n+1)
    level[start]=0
    while q:
            current = q.pop(0)
            for child in graph[current]:
                if visited[child]==0:
                         level[child]=level[current]+1
                         q.append(child)
                         visited[child]=1
            visited[current] = 1

    for i in xrange(1,n+1):
        if i!=start:
            if level[i]!=-1:
                print 6*level[i],
            else:
                print -1,
    print             
